Scripts/Modules/TooltipModule/TooltipModule.cs

```diff
@@ -91,21 +94,28 @@ void Update()
 
 				if (!IsValidTooltip(tooltip))
 					m_TooltipsToHide.Add(tooltip);
+
+				if (tooltipData.persistent)
```

Comment:
Not quite the same manner. Persistent tooltips will only be hidden if duration > 0. Non-persistent tooltips will be hidden as soon as a RayExit occurs, because HideTooltip is called without the persistent flag. The whole point of a persistent tooltip is that it won't be hidden by ray exits. Similarly, we |= persistent, and only set its duration if >0 so that Ray Enters don't unset persistent states.

Reply:
The follow up question is why highlights don't need to follow this pattern as well?

Reply:
They aren't triggered on hover for anything but scene objects. Maybe they should show up on hover, for the buttons but I didn't think that was necessary at this stage